integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/ContentDisposition.java

```diff
@@ -0,0 +1,261 @@
+        String type = (index >= 0 ? headerValue.substring(0, index) : headerValue).trim();
+        if (type.isEmpty()) {
+            throw new IllegalArgumentException("Content-Disposition header must not be empty");
+        }
+        List<String> parts = new ArrayList<>();
+        parts.add(type);
+        if (index >= 0) {
+            do {
+                int nextIndex = index + 1;
+                boolean quoted = false;
+                while (nextIndex < headerValue.length()) {
+                    char ch = headerValue.charAt(nextIndex);
+                    if (ch == ';') {
+                        if (!quoted) {
+                            break;
+                        }
+                    } else if (ch == '"') {
+                        quoted = !quoted;
+                    }
+                    nextIndex++;
+                }
+                String part = headerValue.substring(index + 1, nextIndex).trim();
+                if (!part.isEmpty()) {
+                    parts.add(part);
+                }
+                index = nextIndex;
+            }
+            while (index < headerValue.length());
+        }
+        return parts;
+    }
+
+    /**
+     * Decode the given header field param as describe in RFC 5987.
+     * <p>Only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported.
+     *
+     * @param input the header field param
+     * @return the encoded header field param
+     * @see <a href="https://tools.ietf.org/html/rfc5987">RFC 5987</a>
+     */
+    private static String decodeHeaderFieldParam(String input) {
+        Assert.notNull(input, "Input String should not be null");
+        int firstQuoteIndex = input.indexOf('\'');
+        int secondQuoteIndex = input.indexOf('\'', firstQuoteIndex + 1);
+        // US_ASCII
+        if (firstQuoteIndex == -1 || secondQuoteIndex == -1) {
+            return input;
+        }
+        Charset charset = Charset.forName(input.substring(0, firstQuoteIndex));
+        Assert.isTrue(UTF_8.equals(charset) || ISO_8859_1.equals(charset),
+                "Charset should be UTF-8 or ISO-8859-1");
+        byte[] value = input.substring(secondQuoteIndex + 1, input.length()).getBytes(charset);
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        int index = 0;
+        while (index < value.length) {
+            byte b = value[index];
+            if (isRFC5987AttrChar(b)) {
+                bos.write((char) b);
+                index++;
+            } else if (b == '%') {
+                char[] array = {(char) value[index + 1], (char) value[index + 2]};
+                bos.write(Integer.parseInt(String.valueOf(array), 16));
+                index += 3;
+            } else {
+                throw new IllegalArgumentException("Invalid header field parameter format (as defined in RFC 5987)");
+            }
+        }
+        return new String(bos.toByteArray(), charset);
+    }
+
+    private static boolean isRFC5987AttrChar(byte c) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "isRFC5987AttrChar" is 18 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)